### PR TITLE
Add Time to 2nd review distribution graph

### PR DESCRIPTION
**Pull request by sebastiancaraballo**
## What does this PR do?
- Add Time to 2nd review distribution graph.
- Refactored & added tests.

Preview:
ezgif com-gif-maker

Resolves [#EM-154](https://rootstrap.atlassian.net/browse/EM-154?atlOrigin=eyJpIjoiYmI1NzhhMjEyNmUxNDRjMTlhZDdiM2FhNzY3OWRiZjkiLCJwIjoiaiJ9)

### CODE REVIEW

**Review by SandroDamilano — APPROVED** [review]
Nice work! 👏

File: app/services/builders/chartkick/project_distribution_data.rb [hunk 1]
```diff
@@ -10,17 +10,19 @@ def call
...
-      def merge_times
-        ::MergeTime.joins(pull_request: :project)
-                   .where(projects: { id: @entity_id })
-                   .where(pull_requests: { merged_at: @query[:value_timestamp] })
+      def metric_name
+        @query[:name]
+      end
+
+      def metric
+        @metric ||= ProjectDistributionDataMetrics.const_get(metric_name.to_s.camelize).new
```
Comment (santib) [inline app/services/builders/chartkick/project_distribution_data.rb:21]: The cons I see with this dynamic approach is that it'd be easy to introduce a bug in the future. Another possible approach would be to have

```ruby
def metric
  @metric ||= case @query[:name]
              when :merge_time
                ProjectDistributionDataMetrics::MergeTime.new
              when :review_turnaround
                ProjectDistributionDataMetrics::ReviewTurnaround.new
              end
end
```

but it's true it's more verbose. So yeah.. I'm ok either way

File: app/services/builders/chartkick/project_distribution_data_metrics/review_turnaround.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module Builders
+  module Chartkick
+    module ProjectDistributionDataMetrics
+      class ReviewTurnaround
+        def retrieve_records(entity_id:, time_range:)
+          ::CompletedReviewTurnaround
+            .joins(review_request: :project)
+            .where(projects: { id: entity_id })
+            .where(created_at: time_range)
```
Comment (santib) [inline app/services/builders/chartkick/project_distribution_data_metrics/review_turnaround.rb:9]: don't we have other attribute more accurate than the `created_at` that's just the creation time of the record in our DB?

**Review by santib — APPROVED** [review]
Good work

File: app/services/builders/chartkick/project_distribution_data.rb (hunk 1 above)
Comment (sebastiancaraballo) [reply app/services/builders/chartkick/project_distribution_data.rb:21]: I'll keep it in mind.

File: app/services/builders/chartkick/project_distribution_data_metrics/review_turnaround.rb (hunk 2 above)
Comment (sebastiancaraballo) [reply app/services/builders/chartkick/project_distribution_data_metrics/review_turnaround.rb:9]: I don't think we do.

**Pull request merged**